src/main/scala/uk/gov/nationalarchives/tdr/api/db/repository/FileRepository.scala

```diff
@@ -1,15 +1,29 @@
 package uk.gov.nationalarchives.tdr.api.db.repository
 
+import java.util.UUID
+
 import slick.jdbc.PostgresProfile.api._
 import uk.gov.nationalarchives.Tables
-import uk.gov.nationalarchives.Tables.{File, FileRow}
+import uk.gov.nationalarchives.Tables.{Avmetadata, Consignment, File, FileRow}
 
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
 
 class FileRepository(db: Database) {
   private val insertQuery = File  returning File.map(_.fileid)into ((file, fileid) => file.copy(fileid = fileid))
 
   def addFiles(fileRows: Seq[FileRow]): Future[Seq[Tables.FileRow]] = {
     db.run(insertQuery ++= fileRows)
   }
+
+  def countFilesInConsignment(consignmentId: UUID): Future[Int] = {
+    val query = File.filter(_.consignmentid === consignmentId).length
+    db.run(query.result)
+  }
+
+  def countProcessedFilesInConsignment(consignmentId: UUID): Future[Int] = {
+    val query = (for {
```

Comment:
While I was debugging the test failure yesterday, I noticed that you can remove the `for` comprehension to simplify this query a bit:

```
val query = Avmetadata.join(File)
      .on(_.fileid === _.fileid)
      .filter(_._2.consignmentid === consignmentId)
      .groupBy(_._1.fileid)
      .map(_._1)
      .length
    db.run(query.result)
```

It now reads a bit more like the SQL query it's based on.